### PR TITLE
[Python] Specify metric type in Dataset.create_index

### PR DESCRIPTION
**Pull request by eddyxu**
Allow user to choose create cosine or l2 index from Python

### CODE REVIEW

**Comment by eddyxu** [comment]
Pending a rebase to #600

File: python/python/lance/dataset.py [hunk 1]
```diff
@@ -317,6 +324,9 @@ def create_index(
                         "Vector ndim must be divisible by 8 for SIMD. "
                         "Set `force_build=True` to continue build anyways."
                     )
+
+        if not isinstance(metric, str) or metric.lower() not in ["l2", "cosine"]:
```
Comment (changhiskhan) [inline python/python/lance/dataset.py:328]: Could also map "euclidean" to "l2" just for convenience

File: python/src/dataset.rs [hunk 2]
```diff
@@ -225,6 +228,16 @@ impl Dataset {
             params.num_sub_vectors = PyAny::downcast::<PyInt>(n)?.extract()?
         };
 
+        params.metric_type = match metric_type.to_lowercase().as_str() {
```
Comment (changhiskhan) [inline python/src/dataset.rs:231]: Could maybe do a FromStr for MetricType?

**Review by changhiskhan — COMMENTED** [review]
In the FAISS code if you used angular distance metric, I remember it would normalize the vectors in the index building process. I wasn't sure whether that was for correctness or performance or smth else

**Comment by eddyxu** [comment]
> In the FAISS code if you used angular distance metric, I remember it would normalize the vectors in the index building process. I wasn't sure whether that was for correctness or performance or smth else

You mean that it does normalization within the indexing process? we can look into that as a follow up.

**Review by changhiskhan — APPROVED** [review]

**Pull request merged**